scripts/install/providers/azdo/azdo.sh

```diff
@@ -133,6 +167,133 @@ function configure_repo {
   _success "Project '${AZDO_PROJECT_NAME}' created."
 }
 
+function configure_runners {
+  # Replace in all files in ./.azure-pipelines "runs-on: ubuntu-latest" with "runs-on: self-hosted"
+  if [[ "$OSTYPE" == "darwin"* ]]; then
+      for file in ./.azure-pipelines/*; do
+          sed -i '' -E 's/([[:space:]]*)vmImage: \$\(agentImage\)([[:space:]]*)/\1name: 'Default'\2/g' "$file"
+      done
+  else
+      for file in ./.azure-pipelines/*; do
+          sed -i -E 's/([[:space:]]*)vmImage: \$\(agentImage\)([[:space:]]*)/\1name: 'Default'\2/g' "$file"
+      done
+  fi
+
+  # Verify that required variables exist
+  local REQUIRED_VARS=(\
+      AZDO_ORG_URI \
+      AZDO_PAT \
+      RUNNERS_RESOURCE_GROUP \
+      RUNNERS_LOCATION \
+      RUNNERS_SUBNET \
+      RUNNERS_PUBLIC_KEY_PATH)
+  for var in "${REQUIRED_VARS[@]}"; do
+      if [[ -z "${!var}" ]]; then
+          _error "The variable '$var' is not defined. Aborting."
+          exit 1
+      fi
+  done
+
+  # Validate that the file exists
+  if [[ ! -f "$RUNNERS_PUBLIC_KEY_PATH" ]]; then
+      _error "The specified public key file '$RUNNERS_PUBLIC_KEY_PATH' does not exist. Aborting."
+      exit 1
+  fi
+
+  # Optional parameter: VM size (default: Standard_D2s_v3)
+  RUNNERS_AZURE_VM_SIZE=${RUNNERS_AZURE_VM_SIZE:-Standard_D2s_v3}
+
+  # Optional parameter: Number of runners to launch (default: 1)
+  RUNNERS_COUNT=${RUNNERS_COUNT:-1}
+
+  # Optional parameter: Image (default: Ubuntu2404)
+  RUNNERS_AZURE_IMAGE=${RUNNERS_AZURE_IMAGE:-Ubuntu2404}
+
+  # Optional parameter: VM name (default: symphony-azdo-runner)
+  RUNNERS_VM_NAME=${RUNNERS_VM_NAME:-symphony-azdo-runner}-${AZDO_PROJECT_NAME}
+
+  # Optional parameter: VM username (default: azureuser)
+  RUNNERS_VM_USERNAME=${RUNNERS_VM_USERNAME:-azureuser}
+
+  local token=$AZDO_PAT
```

Comment:
nit: local keyword in bash may break some linters (especially if using strict sh mode)

Reply:
Agree. However symphony is full of `local`. If that's a problem, it would be a problem all over the codebase.